features/gebruik_in_lopende_tekst.feature

```diff
@@ -1,189 +1,651 @@
+
+    Voorbeelden:
+      | adellijkeTitelPredikaat partner | omschrijving | gebruikInLopendeTekst           |
+      | B                               | Baron        | barones van den Aedel-de Boer   |
+      | G                               | Graaf        | gravin van den Aedel-de Boer    |
+      | H                               | Hertog       | hertogin van den Aedel-de Boer  |
+      | M                               | Markies      | markiezin van den Aedel-de Boer |
+      | P                               | Prins        | prinses van den Aedel-de Boer   |
+      | R                               | Ridder       | mevrouw Van den Aedel-de Boer   |
+      | JH                              | Jonkheer     | mevrouw Van den Aedel-de Boer   |
+      | BS                              | Barones      | mevrouw Van den Aedel-de Boer   |
+
+  Abstract Scenario: persoon met geslacht "<geslacht>" en partner met geslacht "<geslacht partner>"
+    Gegeven het systeem heeft een persoon met de volgende gegevens
+    | naam                | waarde     |
+    | burgerservicenummer | 999992934  |
+    | geslachtsaanduiding | <geslacht> |
+    En de persoon heeft de volgende naam gegevens
+    | naam                                 | waarde        |
+    | voornamen (02.10)                    | Jo            |
+    | adellijke titel of predikaat (02.20) |               |
+    | voorvoegsel (02.30)                  | de            |
+    | geslachtsnaam (02.40)                | Boer         |
+    | aanduidingNaamgebruik (61.10)        | P             |
+    En de persoon heeft een partner met de volgende gegevens
+    | naam                        | waarde             |
+    | geslachtsaanduiding (04.10) | <geslacht partner> |
+    En de partner heeft de volgende naam gegevens
+    | naam                                 | waarde                            |
+    | adellijke titel of predikaat (02.20) | B                                 |
+    | voorvoegsel (02.30)                  | van den                           |
+    | geslachtsnaam (02.40)                | Aedel                             |
+    Als personen wordt gezocht met de volgende parameters
+    | naam                | waarde                          |
+    | type                | RaadpleegMetBurgerservicenummer |
+    | burgerservicenummer | 999992934                       |
+    | fields              | naam.gebruikInLopendeTekst                     |
+    Dan bevat de persoon met burgerservicenummer '999992934' de volgende 'naam' gegevens
+    | naam                  | waarde                  |
+    | gebruikInLopendeTekst | <gebruikInLopendeTekst> |
+
+    Voorbeelden:
+      | geslacht | geslacht partner | gebruikInLopendeTekst |
+      | V        | M                | barones van den Aedel |
+      | V        | V                | mevrouw Van den Aedel |
+      | V        | O                | mevrouw Van den Aedel |
+      | V        |                  | mevrouw Van den Aedel |
```

Comment:
Kan het voorkomen dat een persoon geen geslacht heeft? Dus niet M, V of O?

Reply:
Niet in de registratie. In theorie wel wanneer de gebruiker niet geautoriseerd is voor geslacht. Nu komt dat niet voor bij de eerste aansluiters op de API.

Voor de pilot hanteren we de regel dat een samengesteld/afgeleid gegeven niet geleverd mag worden wanneer de gebruiker niet geautoriseerd is voor een van de gegevens waaruit het wordt samengesteld/afgeleid. Voor geslachtsaanduiding van de partner hebben we daar al een uitzondering op gemaakt (geslacht partner null wordt afgehandeld als geslacht partner onbekend). Voor geslacht van de persoon zelf zal dat voorlopig niet spelen. 

Op termijn (na de pilot) worden verwacht ik dat er specifieke autorisaties voor gebruik van de Haal Centraal API worden vastgesteld. Een gebruiker die bijvoorbeeld alleen de partner nodig heeft om de naam samen te kunnen stellen kan dan wel geautoriseerd worden voor aanschrijfwijze, aanhef en gebruikInLopendeTekst, maar niet meer voor partner.

Reply:
Even checken of ik dit goed begrijp. Voor aanschrijfwijze zijn de naam gegevens van de partner nodig. Als de gebruiker niet is geautoriseerd voor de naam gegevens van de partner, dan is dit voor aanschrijfwijze hetzelfde als 'nooit een actuele/ontbonden partner' scenario, en wordt dan aanduiding naamgebruik "E" gehanteerd ipv niet leveren van samengesteld gegeven. Klopt dit?

Reply:
> Klopt dit?

Gelukkig zijn alle (pilot) aansluiters geautoriseerd voor de naamgegevens van partners. 

Niet alle aansluiters zijn geautoriseerd voor de geslachtsaanduiding van de partner. 

Hiervoor hangt het er vanaf hoe we omgaan met autorisatie. Volgens mij hebben we eerder afgesproken dat wanneer met fields gevraagd wordt om een veld waarvoor de gebruiker niet geautoriseerd is, de provider (RvIG) een foutmelding (403?) geeft.

Dat is niet handig, want hoe weet de proxy of gevraagd mag worden om de geslachtsaanduiding van de partner?

Wanneer we anders besluiten, dat vragen om gegevens waarvoor je niet geautoriseerd bent die gegevens niet levert, is dat probleem voor geslachtsaanduiding partner opgelost.

In dat geval zou dan inderdaad bij een aansluiter die niet geautoriseerd is voor partnergegevens de proxy geen verschil zien tussen een persoon die nooit een partner heeft gehad en niet geautoriseerd zijn voor partnergegevens. Gelukkig zijn alle (pilot) aansluiters geautoriseerd voor de naamgegevens van partners.